src/SpendPermissionManager.sol

```diff
@@ -488,14 +496,22 @@ contract SpendPermissionManager is EIP712 {
         // transfer tokens from account to recipient
         if (token == NATIVE_TOKEN) {
             _execute({account: account, target: recipient, value: value, data: hex""});
-        } else {
+            return;
+        }
+        // set infinite allowance if not yet set
+        uint256 allowance = IERC20(token).allowance(account, address(this));
+        if (allowance != type(uint256).max) {
             _execute({
                 account: account,
                 target: token,
                 value: 0,
-                data: abi.encodeWithSelector(IERC20.transfer.selector, recipient, value)
+                data: abi.encodeWithSelector(IERC20.approve.selector, address(this), type(uint256).max)
             });
         }
+
+        // use ERC-20 allowance to transfer from account to recipient
+        bool success = IERC20(token).transferFrom(account, recipient, value);
+        if (!success) revert ERC20TransferFailed(token, account, recipient, value);
```

Comment:
I'm actually second guessing this exact approach. Can you confirm  what `success` is if no return data is returned? I recall SafeTransfer logic typically only enforces `success` if there is actually data returned. Some non-compliant will not return data even if transfer is successful so it's not obvious how that is handled here. Can we write a test with 3 kind of mock erc20s to get an exact verification of the behavior:
1. reverts on failure, returns true otherwise
2. reverts on failure, does not return any data
3. does not revert, returns true/false

Reply:
this is such a good callout.  I was able to produce the problematic state (an ERC20 that doesn't return anything on success) and write a test that fails because our function reverts.

I think the path forward with this is to use OZ's `SafeERC20` for the transfer, which handles the logic of interpreting the return value (or handling the missing return value) that we were attempting roll ourselves.